### PR TITLE
Rule 6.1.4 several init blocks are redundant and generally should not be used in your class

### LINKED ISSUE #440: Rule 6.1.4: several init blocks are redundant and generally should not be used in your class
The primary constructor cannot contain any code. That's why Kotlin has introduced `init` blocks.
These blocks are used to store the code that should be run during the initialization of the class.
Kotlin allows to write multiple initialization blocks that are executed in the same order as they appear in the class body.
Even when you have the (rule 3.2)[#s3.2] this makes code less readable as the programmer needs to keep in mind all init blocks and trace the execution of the code.
So in your code you should try to use single `init` block to reduce the complexity. In case you need to do some logging or make some calculations before the assignment 
of some class property - you can use powerful functional programming. This will reduce the possibility of the error, when occasioanlly someone will change the order of your `init` blocks. 
And it will make the logic of the code more coupled. It is always enough to use one `init` block to implement your idea in Kotlin.

Bad example A:
```kotlin
class YourClass(var name: String) {    
    init {
        println("First initializer block that prints ${name}")
    }
    
    val property = "Property: ${name.length}".also(::println)
    
    init {
        println("Second initializer block that prints ${name.length}")
    }
}
```

Good example A:
```kotlin
class YourClass(var name: String) {
    init {
        println("First initializer block that prints ${name}")
    }

    val property = "Property: ${name.length}".also { prop ->
        println(prop)
        println("Second initializer block that prints ${name.length}")
    }
}
```

Also - init block was not added to Kotlin to help you simply initialize your properties it is needed for more conplex tasks. 
So if `init` block contains only assignments of variables - move it directly to properties so they will be correctly initialized near the declaration.
In some case this rule can be in clash with [6.1.1](https://github.com/cqfn/diKTat/issues/429), but that should not stop you.
Bad example:
```kotlin
class A(baseUrl: String) {
    private val customUrl: String
    init {
        customUrl = "$baseUrl/myUrl"
    }
}
```

Good example:
```kotlin
class A(baseUrl: String) {
    private val customUrl = "$baseUrl/myUrl"
}
```

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Initial implementation

This pull request closes #440

## Actions checklist
* [x] Implemented Rule, added Warnings
* [x] Added tests on checks
* [x] Added tests on fixers
* [x] Updated diktat-analysis.yml
* [x] Updated available-rules.md

## Fixme
* [x] More complicated fixing logic when class properties are assigned inside `init` blocks

### CODE REVIEW

File: info/diktat-kotlin-coding-style-guide-en.md [hunk 1]
```diff
@@ -1769,8 +1769,8 @@ class YourClass(var name: String) {
 }
 ```
 
-Also - init block was not added to Kotlin to help you simply initialize your properties it is needed for more conplex tasks. 
-So if `init` block contains only assignments of variables - move it directly to properties so they will be correctly initialized near the declaration.
+Also - init block was not added to Kotlin to help you simply initialize your properties it is needed for more complex tasks. 
+So if `init` block contains only assignments of variables - move it directly to properties, so they will be correctly initialized near the declaration.
```
Comment (aktsay6) [inline info/diktat-kotlin-coding-style-guide-en.md:1773]: And what about this? Should we consider this logic it rule?

Comment (petertrr) [reply info/diktat-kotlin-coding-style-guide-en.md:1773]: Yes, I'm doing it right now.

**Review by aktsay6 — APPROVED** [review]
lgtm

File: diktat-rules/src/test/resources/test/chapter6/init_blocks/InitBlocksTest.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package test.chapter6.init_blocks
+
+class Example {
+    init { println("Lorem ipsum") }
```
Comment (akuleshov7) [inline diktat-rules/src/test/resources/test/chapter6/init_blocks/InitBlocksTest.kt:4]: @petertrr I have started thinking right now that there is a possibility to break the code here. 
If someone will do the following:
```
init { println(1) }
    val a = "".also {
        println(3)
    } 
init { println(2) }
```

We will break expected print chain

**Pull request merged**